provider/clickhouse.go

```diff
@@ -0,0 +1,1305 @@
+	}
+	columnNames, err := store.query.getColumns(store.db, name)
+
+	return &clickhousePrimaryTable{
+		db:     store.db,
+		name:   name,
+		schema: TableSchema{Columns: columnNames},
+		query:  store.query,
+	}, nil
+}
+
+func (store *clickHouseOfflineStore) UpdateTransformation(config TransformationConfig) error {
+	name, err := store.createTransformationName(config.TargetTableID)
+	if err != nil {
+		return err
+	}
+	err = store.query.transformationUpdate(store.db, name, config.Query)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (store *clickHouseOfflineStore) CreatePrimaryTable(id ResourceID, schema TableSchema) (PrimaryTable, error) {
+	if err := id.check(Primary); err != nil {
+		return nil, err
+	}
+	if exists, err := store.tableExists(id); err != nil {
+		return nil, err
+	} else if exists {
+		return nil, &TableAlreadyExists{id.Name, id.Variant}
+	}
+	if len(schema.Columns) == 0 {
+		return nil, fmt.Errorf("cannot create primary table without columns")
+	}
+	tableName, err := GetPrimaryTableName(id)
+	if err != nil {
+		return nil, err
+	}
+	table, err := store.newsqlPrimaryTable(store.db, tableName, schema)
+	if err != nil {
+		return nil, err
+	}
+	return table, nil
+}
+
+func (store *clickHouseOfflineStore) GetPrimaryTable(id ResourceID) (PrimaryTable, error) {
+	name, err := GetPrimaryTableName(id)
+	if err != nil {
+		return nil, err
+	}
+	if exists, err := store.tableExists(id); err != nil {
+		return nil, err
+	} else if !exists {
+		return nil, &TableNotFound{id.Name, id.Variant}
+	}
+	columnNames, err := store.query.getColumns(store.db, name)
+
+	return &sqlPrimaryTable{
+		db:     store.db,
+		name:   name,
+		schema: TableSchema{Columns: columnNames},
+		query:  store.query,
+	}, nil
+}
+
+func (store *clickHouseOfflineStore) CreateResourceTable(id ResourceID, schema TableSchema) (OfflineTable, error) {
+	if err := id.check(Feature, Label); err != nil {
+		return nil, fmt.Errorf("ID check failed: %v", err)
+	}
+
+	if exists, err := store.tableExists(id); err != nil {
+		return nil, fmt.Errorf("could not check if table exists: %v", err)
+	} else if exists {
+		return nil, &TableAlreadyExists{id.Name, id.Variant}
+	}
+	tableName, err := store.getResourceTableName(id)
+	if err != nil {
+		return nil, fmt.Errorf("could not get resource table name: %v", err)
+	}
+	var valueType ValueType
+	if valueIndex := store.getValueIndex(schema.Columns); valueIndex > 0 {
+		valueType = schema.Columns[valueIndex].ValueType
+	} else {
+		valueType = NilType
+
+	}
+	table, err := store.newsqlOfflineTable(store.db, tableName, valueType)
+	if err != nil {
+		return nil, fmt.Errorf("could not return SQL offline table: %v", err)
+	}
+	return table, nil
+}
+
+func (store *clickHouseOfflineStore) GetResourceTable(id ResourceID) (OfflineTable, error) {
+	return store.getsqlResourceTable(id)
+}
+
+func (store *clickHouseOfflineStore) GetBatchFeatures(ids []ResourceID) (BatchFeatureIterator, error) {
```

Comment:
This can be implemented similar to [SQL GetBatchFeatures](https://github.com/featureform/featureform/blob/main/provider/sql.go#L581-L652)

Reply:
added